myia/prim/shape_inferrers.py

```diff
@@ -439,3 +466,22 @@ def to_element(self, sh):
     ty = await shpy['type']
     n = max(len(tx.elements), len(ty.elements))
     return TupleShape([NOSHAPE] * n)
+
+
+@shape_inferrer(P.make_list, nargs=None)
+async def infer_shape_make_list(track, *elems):
+    """Infer the return shape of make_list."""
+    shps = [await e['shape'] for e in elems]
+    if len(shps) == 0:
+        return ListShape(NOSHAPE)  # pragma: no cover
```

Comment:
This is very easy to test.

Reply:
Yeah, I took it out, though, because the shape of `[]` is kind of undefined, not `ListShape(NOSHAPE)`, in the sense that if you did `[array1] + []` the result would have shape `ListShape(shape(array1))` -- the shape of [] is just ignored. Better just forbid it for now, I think.